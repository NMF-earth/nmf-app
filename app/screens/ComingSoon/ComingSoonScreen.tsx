import React from "react";
import { ScrollView, View, Image } from "react-native";

import { Text, SocialMedia } from "../../components";
import ImagesAssets from "../../constants/ImagesAssets";
import styles from "./ComingSoonScreen.styles";
import { t, platform } from "../../utils";
import navigationOptions from "./ComingSoonScreen.navigationOptions";

const ComingSoonScreen = () => {
  return (
    <ScrollView style={styles.container}>
      <View style={styles.imageView}>
        <Image
          style={styles.image}
          resizeMode="contain"
          source={ImagesAssets.stickers.bike}
        />
      </View>
      <View style={styles.textView}>
        <Text.Primary style={styles.paragraph}>
          {t("COMING_SOON_SCREEN_UNDER_DEVELOPMENT")}
        </Text.Primary>
        <Text.Primary style={styles.paragraph}>
          {t("COMING_SOON_SCREEN_START_FOLLOW_US")}
        </Text.Primary>
      </View>
      <SocialMedia />
    </ScrollView>
  );
};

ComingSoonScreen.navigationOptions = platform.isAndroid
  ? null
  : navigationOptions;

export default ComingSoonScreen;
