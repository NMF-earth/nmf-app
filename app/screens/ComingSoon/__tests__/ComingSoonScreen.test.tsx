import React from "react";
import renderer from "react-test-renderer";
import ComingSoonScreen from "../ComingSoonScreen";

it("ComingSoonScreen renders correctly", () => {
  const tree = renderer.create(<ComingSoonScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
