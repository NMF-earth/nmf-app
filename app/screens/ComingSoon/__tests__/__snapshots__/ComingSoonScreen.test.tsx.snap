// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComingSoonScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "height": "100%",
          "marginTop": 14,
          "width": "100%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 650,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginTop": 22,
        }
      }
    >
      <Text.Primary
        style={
          Object {
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        This feature is under development and will be available soon.
      </Text.Primary>
      <Text.Primary
        style={
          Object {
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        In the mean time, follow us on social media!
      </Text.Primary>
    </View>
    <SocialMedia />
  </View>
</RCTScrollView>
`;
